### PR TITLE
[Fixed] - Resolve Add Item To Wish-list With Default Value Below 1.

### LINKED ISSUE #27396: Magento 2.3.4 ad item to wishlist with default value below 1
### Preconditions (*)

1. Magento 2.3.4

### Steps to reproduce (*)

1. Add product with default value 0,5 (eg. meters) - Minimum Qty Allowed in Shopping Cart
2. Log in with account in frontend
3. Go to product and add to wishlist

### Expected result (*)

1.  Quantity in wishlist is 0,5

### Actual result (*)

1. Quantity in wishlist is 1

stap1
stap2

### PR DESCRIPTION
**Pull request by niravkrish**
### Description (*)

- I have added minimum saleable quantity as a default qty add to wishlist on add to wishlist button params.

### Fixed Issues (if relevant)

1. magento/magento2#27396: Magento 2.3.4 ad item to wishlist with default value below 1

### Manual testing scenarios (*)

1. Add to wishlist from different page like from homepage, Listing page and details page.
2. Change qty and add to wishlist.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Wishlist/Helper/Data.php [hunk 1]
```diff
@@ -145,6 +153,7 @@ public function __construct(
         $this->wishlistProvider = $wishlistProvider;
         $this->productRepository = $productRepository;
         $this->escaper = $escaper ?? ObjectManager::getInstance()->get(Escaper::class);
```
Comment (lbajsarowicz) [inline app/code/Magento/Wishlist/Helper/Data.php]:
```suggestion
        $this->escaper = $escaper;
```

File: app/code/Magento/Wishlist/Helper/Data.php [hunk 2]
```diff
@@ -145,6 +153,7 @@ public function __construct(
...
+        $this->stockItemRepository = $stockItemRepository;
```
Comment (lbajsarowicz) [inline app/code/Magento/Wishlist/Helper/Data.php]:
```suggestion
        $this->stockItemRepository = $stockItemRepository ?? ObjectManager::getInstance()->get(StockItemRepository::class);
```

File: app/code/Magento/Wishlist/Helper/Data.php [hunk 3]
```diff
@@ -134,7 +141,8 @@ public function __construct(
         \Magento\Customer\Helper\View $customerViewHelper,
         WishlistProviderInterface $wishlistProvider,
         \Magento\Catalog\Api\ProductRepositoryInterface $productRepository,
-        Escaper $escaper = null
+        Escaper $escaper = null,
+        StockItemRepository $stockItemRepository
```
Comment (lbajsarowicz) [inline app/code/Magento/Wishlist/Helper/Data.php]: Remove backward compatible Object loader for escaper, introduce such for your new dependency.


```suggestion
        Escaper $escaper,
        StockItemRepository $stockItemRepository = null
```

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Please review the comments.

**Comment by lbajsarowicz** [comment]
Let's wait for tests to pass now :-)
Make sure that Unit Tests and Integration Tests are running :heavy_check_mark: for you

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
Working with MFTF test

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
Some anomaly with tests.

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@VladimirZaets 
Seems Tests can not be run.
Issue with Tests by Test Server? Or by changes in Pull Request?

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@niravkrish 
Please check failed Tests.

**Comment by lbajsarowicz** [comment]
@niravkrish closing due to inactivity. Please let me know if you still want to work on the issue.

**Pull request closed without merging**